pyproject.toml

```diff
@@ -0,0 +1,121 @@
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[tool.hatch.version]
+source = "vcs"
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/s3_log_extraction"]
+
+[project]
+name = "s3_log_extraction"
+version="0.0.1"
+authors = [
+  { name="Cody Baker", email="cody.c.baker.phd@gmail.com" },
+]
+description = "Fast extraction of access summary data for S3 logs."
+readme = "README.md"
+keywords = ["aws", "s3", "log", "download tracking", "geolocation"]
+license = {file = "license.txt"}
+requires-python = ">=3.12"
+dependencies = [
+    "pandas",
+    "tqdm",
+    "ipinfo",
+    "opencage",
+    "PyYAML",
+    "click",
+    "natsort",
+    "dandi",
+    "cryptography",
+]
+classifiers = [
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.12",
+    "License :: OSI Approved :: BSD License",
+    "Intended Audience :: Developers",
+    "Operating System :: Unix",
+]
+
+#[project.scripts]
+#reduce_all_dandi_raw_s3_logs = "s3_log_extraction._command_line_interface:_reduce_all_dandi_raw_s3_logs_cli"
+#bin_all_reduced_s3_logs_by_object_key = "s3_log_extraction._command_line_interface:_bin_all_reduced_s3_logs_by_object_key_cli"
+#map_binned_s3_logs_to_dandisets = "s3_log_extraction._command_line_interface:_map_binned_s3_logs_to_dandisets_cli"
+#generate_dandiset_summaries = "s3_log_extraction._command_line_interface:_generate_dandiset_summaries_cli"
+#generate_all_dandiset_totals = "s3_log_extraction._command_line_interface:_generate_all_dandiset_totals_cli"
+#generate_archive_summaries = "s3_log_extraction._command_line_interface:_generate_archive_summaries_cli"
+#generate_archive_totals = "s3_log_extraction._command_line_interface:_generate_archive_totals_cli"
+#update_region_codes_to_coordinates = "s3_log_extraction._command_line_interface:_update_region_codes_to_coordinates_cli"
+#check_for_errors = "s3_log_extraction._command_line_interface:_check_for_errors"
+
+[project.optional-dependencies]
+dev = [
```

Comment:
I think we can benefit from hatch's support for environments. We can define an environment just for development/testing instead of defining a feature/extra with the name of `dev`, see https://hatch.pypa.io/latest/config/environment/overview/#dependencies for details.

Reply:
Good to know - could you show what that would look like in a separate PR so I can test/play around with it?

I guess all I really care about is that it plays well with `pip install[ whatever other extras ]` syntax

Hatch looks very node-inspired (even has pre- and post- stages!), very cool